lib/rules/no-implicit-coercion.js

```diff
@@ -245,28 +247,50 @@ module.exports = {
          * @returns {void}
          */
         function report(node, recommendation, shouldFix) {
+
+            /**
+             * Fix function
+             * @param {RuleFixer} fixer The fixer to fix.
+             * @returns {Fix} The fix object.
+             */
+            function fix(fixer) {
+                const tokenBefore = sourceCode.getTokenBefore(node);
+
+                if (
+                    tokenBefore &&
+                    tokenBefore.range[1] === node.range[0] &&
+                    !astUtils.canTokensBeAdjacent(tokenBefore, recommendation)
+                ) {
+                    return fixer.replaceText(node, ` ${recommendation}`);
+                }
+
+                return fixer.replaceText(node, recommendation);
+            }
+
             context.report({
                 node,
-                messageId: "useRecommendation",
-                data: {
-                    recommendation
-                },
+                messageId: "implicitCoercion",
+                data: { recommendation },
                 fix(fixer) {
                     if (!shouldFix) {
                         return null;
                     }
 
-                    const tokenBefore = sourceCode.getTokenBefore(node);
-
-                    if (
-                        tokenBefore &&
-                        tokenBefore.range[1] === node.range[0] &&
-                        !astUtils.canTokensBeAdjacent(tokenBefore, recommendation)
-                    ) {
-                        return fixer.replaceText(node, ` ${recommendation}`);
+                    return fix(fixer);
+                },
+                suggest: [
+                    {
+                        messageId: "useRecommendation",
+                        data: { recommendation },
+                        fix(fixer) {
+                            if (shouldFix) {
+                                return null;
+                            }
```

Comment:
[Question] Is it unusual to have a fixer or suggestion always `return null` based on some condition outside the fixing function? As in, would any downstream tooling be inconvenienced by falsely thinking these functions would do something?

Even if there's no functional difference, I did find it confusing to see a full `suggest` array. That feels slightly more unusual than the existing code's `if (!shouldFix)` condition being inside its `fix`. My _hunch_ is that it'd be better to conditionally assign `fix` and `suggest` only if they should exist.

```js
context.report({
   /* ... */
   ...(shouldFix ? { fix } : { suggest: [/* ... */] })
})
```

...but I'm new here and not confident in this suggestion.

Reply:
Functions included in the `suggest` array are not exposed directly to API consumers. Suggestions are processed by the linter to format the description message and normalize the fix ([here](https://github.com/eslint/eslint/blob/39076fb5e4c7fa10b305d510f489aff34a5f5d99/lib/linter/report-translator.js#L217-L229)), and only the mapped objects are made available. In this process, suggestions without a fix, or with a `fix` function that returns `null` are filtered out. Same thing for fixes: a `fix` function is invoked in the linter and only its results are exposed.